YololEmulator.Tests/Analysis/Inspection/FindBooleans.cs

```diff
@@ -31,10 +31,52 @@ public void FindBooleansSingleLine()
                 new VariableName("f[0]"),
                 new VariableName("b[0]")
                 }));
+        }
+
+        [TestMethod]
+        public void FindBooleanLiterals()
+        {
+            var ast = TestExecutor.Parse(
+                "a=1 theAnswer=42 b=theAnswer==a"
+            );
+
+            var cfg = new Yolol.Analysis.ControlFlowGraph.Builder(ast.StripTypes()).Build();
+
+            cfg = cfg.StaticSingleAssignment(out var ssa);
+
+            var variableNames = cfg.FindBooleanVariables(ssa).ToHashSet();
+
+            Assert.IsTrue(variableNames.SetEquals(new VariableName[] {
```

Comment:
This test relies on the naming scheme of the AST builder, which is an unstable implementation detail. It's not critical to fix that in this PR.

Reply:
I'm taking a look at it now. I feel like I might have a solution.

Reply:
Btw the naming of `a[0]` and `b[0]` ought to be stable, it's just the intermediate values that may change (that's dependent upon whatever the ExpressionDecomposition pass does). So you could just validate that `a[0]` and `b[0]` get classified as booleans.